packages/quicktype-core/src/language/CPlusPlus.ts

```diff
@@ -508,11 +501,13 @@ export class CPlusPlusRenderer extends ConvenienceRenderer {
 
         if (_options.boost) {
             this._optionalType = "boost::optional";
+            this._optionalFactory = "boost::optional";
             this._nulloptType = "boost::none";
             this._variantType = "boost::variant";
             this._variantIndexMethodName = "which";
         } else {
             this._optionalType = "std::optional";
+            this._optionalFactory = "std::make_optional";
```

Comment:
Please ensure the test suite actually exercises this on a few files.

Reply:
I think you will find it tests this on many files. Any files that previously had optionals in the schema were using boost::shared pointer. They now use boost::optional. If C++17 is selected they will used std::shared_ptr and std::optional.

The feature is available by default as requested

Reply:
However I'll do a quick grep through the test outputs directory to see if I can identify how many files are triggering boost::optional members

Reply:
The attached file shows that boost::optional is generated is many test cases and also that std::shard_ptr is also generated in a subset of them. These are part of the round tripping test suite so is this good enough to prove the code is being excercised?

[optional.txt](https://github.com/quicktype/quicktype/files/10443618/optional.txt)